Documentation Updates!

sheryjoe opened the pull request:
This PR includes the resolution of #649 #650, #648 and MORE and MORE, see below.  

What is left is 
- the documentation for the cutting plane use case @HeavenlyBerserker please provide needed information as mentioned in my email.
- feature map documentation in "what is new", #712
- @akenmorris please watch out for any edits needed in light of new Studio updates in #712 


This PR includes the following:
- Consolidated mkdocs.md and autodoc.md -> docs.md.  
- Rewrote analyze.md and optimize.md with more discussions and illustrations
- Added latex support to our docs for equations so we can use latex and render it very nicely (as opposed to low-res images of equations).
- Added support for call-outs and added instructions to use it in docs.md
- Additions to the home page
    - Added announcement to the get-togethers
    - Why ShapeWorks?
    - ShapeWorks vs other tools on a simple example
- Added "What is new" section with illustrations for 
    - dynamic loading (in Studio)
    - surface reconstruction (in Studio)
    - particles on meshes
    - consolidated commands
    - memory optimizations
    - Live particles movement (in Studio)
    - Quantitative evaluation for shape models 
- Added selected success stories to demonstrate ShapeWorks impact
- Some theme customization, e.g., colored font for sections for more readability
- Fixed DocumentationUtils and added runnable script to avoid having to auto-generate documentation within a python shell
- Added team picture


**Steps:**

- checkout mkdocs branch
- run mkdocs serve from the root directory (containing mkdocs.yml)
- open and review the following:    
    - http://127.0.0.1:8000/dev/
    - http://127.0.0.1:8000/getting-started/sw-stories/
    - http://127.0.0.1:8000/new/new-studio/
    - http://127.0.0.1:8000/new/openvdb/
    - http://127.0.0.1:8000/new/sw-meshes/
    - http://127.0.0.1:8000/new/shapeworks-api/
    - http://127.0.0.1:8000/new/ssm-eval/
    - http://127.0.0.1:8000/dev/docs/
    - http://127.0.0.1:8000/dev/workflow/analyze/
    - http://127.0.0.1:8000/dev/workflow/optimize/
    - http://127.0.0.1:8000/about/team/

**Note:** mkdocs branch will be used for editing/adding docs and I added a branch rule to prevent its deletion.

sheryjoe commented:
@akenmorris please review and test the packages scripts as I changed the autodoc.

riddhishb commented on docs/dev/contribute.md:
```diff
@@ -6,7 +6,8 @@
 
-:fa-info-circle: _**Important:**_ If you want to contribute, first _make sure you have a developer account_.  
+!!! imporant
+    If you want to contribute, first _make sure you have a developer account_.  
```
spell check

riddhishb commented on docs/new/ssm-eval.md:
```diff
@@ -0,0 +1,41 @@
+
+!!! note "Specificity"
+    This is the requirement that the model can only represent valid instances of the class(es) of objects presented in the training set. Hence, the model is specific for this training set.
+
```
Should we add something about what is a good balance between compactness, generalization and specificity as there is a tradeoff between these numbers, the example that compact model =0.99 is misleading a little.

sheryjoe replied:
Good point. Added

riddhishb commented on docs/workflow/optimize.md:
```diff
@@ -1,104 +1,192 @@
- . The optimization to establish correspondence minimizes the energy function
+## Particle-based Shape Modeling
+
+ShapeWorks constructs *statistically optimal anatomical mapping across different shape samples* by automatically computing a dense set of corresponding landmark positions on a set of anatomy segmentations or surface meshes and *does not rely on any specific surface parameterization*. 
```
add geometrically consistent in this sentence basically define what is "in correspondence"

sheryjoe replied:
done

riddhishb reviewed (commented):
Very minor comments in overall clarification but looks good to go otherwise to me

medakk reviewed (approved):
Looks great! I don't know why but the links in the PR (like http://127.0.0.1:8000/dev/) did not work for me. I used to sidebar to get to the pages

sheryjoe commented on docs/dev/build.md:
```diff
@@ -86,7 +90,8 @@ Download and install [[Visual Studio 2019]](https://visualstudio.microsoft.com/)
 #### Anaconda
 Download and install [[Anaconda]](https://www.anaconda.com/). 
```
@akenmorris @cchriste will miniconda be sufficient for our needs here?

sheryjoe commented:
@jadie1 @medakk @akenmorris @cchriste @archanasri @riddhishb @HeavenlyBerserker 

I have addressed all comments, restructured release notes, and added documentation for the cutting planes use cases. PLEASE review and let me know your comments and edits. 

This should be merged VERY SOON.

medakk commented on docs/about/release-notes.md:
```diff
@@ -1,5 +1,69 @@
+
+* **ShapeWorks directly on meshes:** ShapeWorks now supports particle optimization directly on triangular surface meshes. This mesh support also enables working with open meshes without additional user inputs. See [ShapeWorks Directly on Meshes](http://sciinstitute.github.io/ShapeWorks/new/sw-meshes) for more details. For exemplar use cases, see [Femur Mesh: SSM directly from meshes](http://sciin ...
+
+* **New ShapeWorks API:** Consolidation of image-based and segmentation-based grooming tools that creates a stable and reusable API making it much easier and more flexible for users to groom their datasets. This includes a full complement of unit tests. See [ShapeWorks API](http://sciinstitute.github.io/ShapeWork/new/shapeworks-api) for more details.
```
typo in url

sheryjoe replied:
Fixed. Thanks!

medakk commented on docs/about/release-notes.md:
```diff
@@ -1,5 +1,69 @@
+
+#### ShapeWorks Back-end
+
+* **Improved constrained particle optimization:** An improved implementation for cutting planes that support single and multiple cutting planes per sample has been added. This can be used in modeling scenarios where statistical modeling/analysis is needed for a region-of-interest on the anatomy/object-class at hand without having to affect the input data. See [Ellipsoid: Cutting Planes](http://sc ...
```
missing text for femur cutting planes link

sheryjoe replied:
Fixed. Thanks!

HeavenlyBerserker reviewed (approved):
Yup

sheryjoe commented:
@jadie1 @medakk @akenmorris @cchriste @archanasri @riddhishb @HeavenlyBerserker

Please let me know your comments on this graphical abstract for 5.5. This is a screen capture so ignore the red line under shapeworks.

<img width="985" alt="image" src="https://user-images.githubusercontent.com/3253036/95643392-221cf180-0a6c-11eb-8dff-4c7e2970415d.png">

sheryjoe commented:
Another version

<img width="1258" alt="image" src="https://user-images.githubusercontent.com/3253036/95643624-90ae7f00-0a6d-11eb-8f17-2f53ee8988cf.png">

medakk commented:
Its a little cluttered IMO, can't tell very well what images are related to each other. Maybe make it list/grid with clear demarcations(or lots of space in between?)

sheryjoe commented:
@medakk good point. how about this one?

<img width="1060" alt="image" src="https://user-images.githubusercontent.com/3253036/95644106-5941d180-0a71-11eb-94b3-2d14cae992db.png">

sheryjoe commented:
> Looks great!

@jadie1 do you approve the PR?

archanasri commented:
@sheryjoe the latest graphic looks better...less cluttered :)

medakk commented:
@sheryjoe yup this looks good! maybe put the "ShapeWorks" under/besides the logo instead of on top of it

sheryjoe commented:
> @sheryjoe yup this looks good! maybe put the "ShapeWorks" under/besides the logo instead of on top of it

Doing that will break the symmetry of the grid :(

akenmorris commented:
Nice!

HeavenlyBerserker commented:
👍

On Fri, Oct 9, 2020, 9:36 PM Alan Morris <notifications@github.com> wrote:

> Nice!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SCIInstitute/ShapeWorks/pull/746#issuecomment-706480067>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACVO6GFPW2G4OVKVGAW3A5LSJ7JETANCNFSM4SEEQ7HA>
> .
>

sheryjoe commented:
Happy merging! Thanks all for your comments and feedback!
